app/controllers/authenticate_controller.rb

```diff
@@ -136,39 +140,68 @@ def authenticate_jwt
     handle_authentication_error(e)
   end
 
+  def logout_oidc
+    params[:authenticator] = "authn-oidc"
+    decode_and_merge_request_body
+    params.permit!
+
+    logout_uri = Authentication::Handler::LogoutHandler.new(
+      authenticator_type: params[:authenticator]
+    ).call(
+      parameters: params.to_hash.symbolize_keys,
+      request_ip: request.ip
+    ) do |authenticator|
+      Authentication::AuthnOidc::V2::Strategies::Logout.new(
+        authenticator: authenticator
+      ).callback(params.to_hash.symbolize_keys)
+    end
+
+    render(json: { oidc_logout_uri: logout_uri })
+  end
+
   def authenticate_oidc
```

Comment:
This is the original "OIDC" functionality which is really just a glorified JWT validation specific to OIDC. If you're going to hijack the `POST` route, I'd recommend splitting this up a bit differently, something like:

```ruby
  def authenticate_oidc
    # Parse JSON request body
    # If refresh token is present, do refresh token workflow
    # Else, do this:
      params[:authenticator] = "authn-oidc"
      input = Authentication::AuthnOidc::UpdateInputWithUsernameFromIdToken.new.(
        authenticator_input: authenticator_input
      )
      # We don't audit success here as the authentication process is not done
    rescue => e
      # At this point authenticator_input.username is always empty (e.g. cucumber:user:USERNAME_MISSING)
      log_audit_failure(
        authn_params: authenticator_input,
        audit_event_class: Audit::Event::Authn::Authenticate,
        error: e
      )
      handle_authentication_error(e)
    else
      authenticate(input)
    end
```